docs/versioning-policy.md

```diff
@@ -0,0 +1,70 @@
+3. The `patch` version number **SHOULD ONLY** be incremented for a release containing **ONLY** bug fixes and no new features.
+4. A release that increments the `minor` or `patch` version **ONLY**, **MUST** be backward compatible.
+5. Incrementing the `minor` version **MUST** reset the `patch` version number to zero.
+6. Incrementing the `major` version **MUST** reset the `minor` and `patch` version number to zero.
+
+### Unavoidable breaking changes
+
+Due to a reliance on the upstream Terraform providers and prior instances where they introduce breaking changes in minor version releases, we cannot guarantee that a minor version bump will never introduce a breaking change. 
+
+All efforts **SHOULD** be made to automate or work around a breaking change to minimize the impact on the end user. 
+
+When it is unavoidable, the details of the breaking change and how to adapt to it **MUST** accompany the new version's release notes.
+
+## Upgrades
+
+### Backward compatibility
+
+Except for [unavoidable breaking changes](#unavoidable-breaking-changes), backward compatibility expects that there **MUST** be no changes required to the Crossplane resource manifests, configurations or infrastructure when upgrading to a new version with a higher minor or patch version number but the same major version number. For example, upgrading from `v1.2.3` to `v1.2.4` or `v1.3.0`.  
+
+Backward compatibility is **NOT** guaranteed when downgrading to a prior version.
+
+### Major version upgrades
+
+A change in the `major` version number **DOES NOT** come with a backward compatibility guarantee.
+
+All breaking changes introduced **MUST** be indicated in the release notes. 
+
+All efforts **SHOULD** be made to automate or work around the breaking change to minimize the impact on the end user. 
+
+## Minimizing the impact of changes
+
+The following mitigation techniques **SHOULD** be considered when introducing breaking changes is unavoidable.
+
+### Use multiple versions in MR CRDs 
+
+Leverage multiple versions in the generated CRDs for breaking change management. If there’s a breaking change in the API of an MR, publish the new API with a new version so that both the old and the new versions will be available in the OpenAPI schema. This allows the utilization of conversion webhooks between the different versions of a CRD and for proper API deprecation and removal.
+
+### Conversion webhooks 
+
+Having the old APIs stored in a generated CRD, as discussed above with multiple CRD versions, allows for leveraging API converters in the form of Kubernetes conversion webhooks. These could help end users with breaking API changes during the provider upgrade. 
+
+The webhooks should be implemented as part of the provider to simplify deployment.
+
+Upjet provides high-level libraries that make writing these converters fast and robust for specific API changes.
```

Comment:
Having never written a conversion webhook, I'm left wondering where to find these libraries, and a link would be great. But if it's obvious to those with more k8s/golang experience where to find them, you don't need to add it on my account alone.